plugin/web/extensions/sidebar/modals/datacatalog/components/content/DatasetsPage/DatasetTree/Folder.tsx

```diff
@@ -19,30 +18,20 @@ const Folder: React.FC<Props> = ({
   isMobile,
   expandAll,
   nestLevel,
-  item,
-  selectedID,
+  nodeKey,
+  expandedKeys,
+  onExpand,
   children,
 }) => {
   const [isOpen, open] = useState(false);
 
-  // TODO: should improve performance later
-  const isChildSelected = useCallback(
-    (
-      item: DataCatalogGroup | DataCatalogItem | (DataCatalogItem | DataCatalogGroup)[],
-    ): boolean => {
-      if (!("children" in item)) return "dataID" in item && item.dataID === selectedID;
-      return item.children.some(child => isChildSelected(child));
-    },
-    [selectedID],
-  );
-
   useEffect(() => {
-    open(() => expandAll || (!!selectedID && isChildSelected(item)));
-  }, [expandAll, isChildSelected, item, selectedID]);
+    open(() => expandAll || expandedKeys.includes(nodeKey));
```

Comment:
This will happen for ALL folders EVERYTIME a key is added to expandedKeys, right?
Please add an if statement `if(expanded || expandedKeys.includes(nodeKey)` so that we don't update the isOpen state so much.

Reply:
I am not sure whether this is necessary or not.
Do you mean something like this:
```ts
  useEffect(() => {
    expandAll || expandedKeys.includes(nodeKey)
      ? open(expandAll || expandedKeys.includes(nodeKey))
      : open(false);
  }, [expandAll, expandedKeys, nodeKey]);
```

Reply:
Done!